omod/src/main/java/org/openmrs/module/atomfeed/web/controller/AtomfeedConfigurationController.java

```diff
@@ -0,0 +1,137 @@
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The Atomfeed configuration controller.
+ */
+@Controller
+@RequestMapping("/module/atomfeed")
+public class AtomfeedConfigurationController {
+
+	protected static final Logger LOGGER = LoggerFactory.getLogger(AtomfeedConfigurationController.class);
+
+	private static final String CONFIGURATION_MODEL = "configuration";
+	private static final String SAVE_CONFIG_SUCCESS = "atomfeed.configuration.json.save.success";
+	private static final String SAVE_CONFIG_ERROR = "atomfeed.configuration.json.save.fail";
+	private static final String ERRORS_INVALID_FILE = "atomfeed.configuration.errors.invalidFile";
+
+	@Autowired
+	private FeedConfigurationService feedConfigurationService;
+
+	/**
+	 * Sets the UI model attributes used to display the atomfeed configuration etc.
+	 *
+	 * @param model injected the page model object
+	 * @param success injected the flag used to choose the type of alert message
+	 * @param alertMessage injected the message used to display an alert message (display the alert if the value isn't null)
+	 * @return the redirect URL to atomfeedConfiguration.jsp
+	 */
+	@RequestMapping("/configuration")
+	public String get(ModelMap model,
+			@RequestParam(value = AtomfeedMessageUtils.SUCCESS_MESSAGE, required = false) boolean success,
+			@RequestParam(value = AtomfeedMessageUtils.ALERT_MESSAGE_MODEL, required = false) String alertMessage) {
+		String configuration =
+				AtomfeedUtils.writeFeedConfigurationToJsonString(feedConfigurationService.getGeneralConfiguration());
+		model.addAttribute(CONFIGURATION_MODEL, configuration);
+		model.addAttribute(AtomfeedMessageUtils.SUCCESS_MESSAGE, success);
+		model.addAttribute(AtomfeedMessageUtils.ALERT_MESSAGE_MODEL, alertMessage);
+		return "/module/atomfeed/atomfeedConfiguration";
+	}
+
+	/**
+	 * Saves the configuration (sent as JSON string) into the server.
+	 * Notifies the user about the result of this operation.
+	 *
+	 * @param model injected the page model object
+	 * @param json injected the JSON representation of the configuration
+	 * @return the redirect URL (if success to /module/atomfeed/atomfeed if not then /module/atomfeed/configuration)
+	 */
+	@RequestMapping(value = "/saveConfiguration", method = RequestMethod.POST)
+	public String post(ModelMap model,
+			@RequestParam("json") String json) {
+		try {
+			feedConfigurationService.saveConfig(json);
+			AtomfeedMessageUtils.successMessage(model, SAVE_CONFIG_SUCCESS);
+			return "redirect:/module/atomfeed/atomfeed.form";
+		} catch (Exception e) {
```

Comment:
can you use less generic exception ? because here you try to catch all exceptions, even NPE

Reply:
Good point, thanks.